tasks/icinga2-api.yml

```diff
@@ -0,0 +1,121 @@
+---
+- name: Icinga 2 master - Check if Public Key Infrastructure has been initialized
+  become: yes
+  stat:
+    path: "{{ i2_ca_file }}"
+  register: ca_certificate
+
+- name: Icinga 2 master - Create Certificate Authority
+  become: yes
+  command: icinga2 pki new-ca
+  when: (groups['masters']|length == 1 or i2_config_master) and not ca_certificate.stat.exists
+
+- name: Icinga 2 master - Synchronize CA Cert between masters (Fetch)
+  become: yes
+  fetch:
+    src: "{{ i2_ca_path }}/{{ item }}"
+    dest: /tmp/trusted_certs/
+    flat: yes
+  loop: ['ca.crt', 'ca.key']
+  when: (groups['masters']|length > 1 and i2_config_master)
+
+- name: Icinga 2 master - Synchronize CA between masters (Ensure ca directory exists)
+  become: yes
+  file:
+    path: "{{ i2_ca_path }}"
+    state: directory
+    owner: "{{ i2_user }}"
+    group: "{{ i2_group }}"
+    mode: 0700
+  when: (groups['masters']|length > 1 and not i2_config_master) and not ca_certificate.stat.exists
+
+- name: Icinga 2 master - Synchronize CA between masters (Copy)
+  become: yes
+  copy:
+    src: /tmp/trusted_certs/{{ item }}
+    dest: "{{ i2_ca_path }}/"
+    owner: "{{ i2_user }}"
+    group: "{{ i2_group }}"
+    mode: 0644
+  loop: ['ca.crt', 'ca.key']
+  when: (groups['masters']|length > 1 and not i2_config_master) and not ca_certificate.stat.exists
+
+- name: Icinga 2 master - Initialize Certificate Directory
+  become: yes
+  file:
+    path: "{{ i2_pki_path }}"
+    state: directory
+    owner: "{{ i2_user }}"
+    group: "{{ i2_group }}"
+    mode: 0700
+
+- name: Icinga 2 master - Create Host Certificate
+  become: yes
+  command: >
+    icinga2 pki new-cert
+    --cn {{ ansible_fqdn }}
```

Comment:
I haven't used this in a while but given that the `ansible_fqdn` may not return a fully qualified domain name (see [this example in the docs](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variables-discovered-from-systems-facts), I'd be hesitant to rely on it. I'd have a configurable variable that defaults to `ansible_fqdn` so that if it resolves to the wrong name the user can fix it.